server/src/weaverbird/backends/pypika_translator/translators/base.py

```diff
@@ -271,6 +273,52 @@ def aggregate(
                 selected_col_names,
             )
 
+    @staticmethod
+    def _pipeline_or_domain_name_or_reference_to_pipeline(
+        pipeline: 'PipelineOrDomainNameOrReference', message: str | None = None
+    ) -> list['PipelineStep']:
+        try:
+            return Pipeline(steps=pipeline).steps
+        except Exception as exc:
+            message = message or f"could not convert {pipeline} to pipeline: {exc}"
+            raise NotImplementedError(message) from exc
+
+    def append(
+        self: Self,
+        *,
+        builder: 'QueryBuilder',
+        prev_step_name: str,
+        columns: list[str],
+        step: "AppendStep",
+    ) -> StepContext:
+        pipelines = [
+            self._pipeline_or_domain_name_or_reference_to_pipeline(pipeline)
+            for pipeline in step.pipelines
+        ]
+        tables: list[str] = []
+        column_lists: list[list[str]] = []
+        for pipeline in pipelines:
+            pipeline_ctx = self.__class__(
+                tables_columns=self._tables_columns, db_schema=self._db_schema_name
+            ).get_query_builder(steps=pipeline, query_builder=builder)
+            tables.append(pipeline_ctx.table_name)
+            column_lists.append(pipeline_ctx.columns)
+            builder = pipeline_ctx.builder
+
+        # NOTE: Not sure if we should run this check or just let the UNION ALL happen...
+        # if not all(cols == column_lists[0] for cols in column_lists[1:]):
+        #     raise ValueError(
+        #         "Columns of all appended datasets must be the same than the current dataset."
+        #         f''' Expected "{', '.join(column_lists[0])}"'''
+        #     )
```

Comment:
Behaviour here needs to be determined. I see three options:
* Leave things as is -> Most SQL-ish
* Uncomment this section -> Safer, more annoying for users
* Check all columns that'll be required in the final dataset by having a look at everyhting we resolved from the different pipelines, and cheat by doing literal selects:
  `SELECT col1_dataset1, NULL AS "col1_dataset2"` -> Here, we'd create a column containing NULLs, so we could append other columns from other datasets to it. 
:warning: I'm not sure this works, need to test it first